src/widget/wstarrating.cpp

```diff
@@ -162,22 +118,16 @@ int WStarRating::starAtPosition(int x) {
 }
 
 void WStarRating::mouseReleaseEvent(QMouseEvent* /*unused*/) {
-    if (!m_pCurrentTrack) {
-        return;
-    }
-
-    m_pCurrentTrack->setRating(m_starRating.starCount());
+    emit ratingChangeRequest(m_starRating.starCount());
 }
 
 void WStarRating::fillDebugTooltip(QStringList* debug) {
     WWidget::fillDebugTooltip(debug);
 
-    QString currentRating = "-";
+    QString currentRating;
+    currentRating.setNum(m_currRating);
```

Comment:
What's the difference between m_currRating and m_starRating? The widget should not store any redundant state that could get inconsistent.

Reply:
`m_starRating.starCount()` is the current value of StarRating, i.e. what you see (for example while hovering).
`m_currRating` is the confirmed value of the track / pending track record value in DlgTrackInfo, after LeaveEvent.

Reply:
Then `m_currRating` is redundant. The widget should maintain only a single value and don't care what it's clients or subscribers are assuming.

Reply:
> Then `m_currRating` is redundant.

Redundant with what?
The current value of `StarRating` (starCount()) just reflects the count of currently painted stars, be it final (clicked, not necessarily accepted by Track or DlgTrackInfo) or temporary (hovering, mouse moving).

Only `m_currRating` is storing the applied / real value and we need that for the COs and to restore the real value when leaving the widget (after mouse move, not clicked).

Reply:
I'll check it later. Only viewed the diff, not the whole code.

Reply:
Nevertheless my comments reveal that meaning and purpose of those member variables is unclear on first sight.